datastore/fertiscan/__init__.py

```diff
@@ -268,3 +275,36 @@ async def get_user_analysis_by_verified(cursor, user_id,verified:bool):
     except Exception as e:
         print(e.__str__())
         raise Exception("Datastore unhandeled error")
+
+
+async def delete_inspection(
+    cursor: Cursor,
+    inspection_id: str | UUID,
+    user_id: str | UUID,
+    container_client: ContainerClient,
+) -> data_inspection.DBInspection:
+    """
+    Delete an existing inspection record and its associated picture set from the database.
+
+    Parameters:
+    - cursor (Cursor): Database cursor for executing queries.
+    - inspection_id (str | UUID): UUID of the inspection to delete.
+    - user_id (str | UUID): UUID of the user performing the deletion.
+
+    Returns:
+    - data_inspection.Inspection: The deleted inspection data from the database.
+
+    """
+    if isinstance(inspection_id, str):
+        inspection_id = UUID(inspection_id)
+    if isinstance(user_id, str):
+        user_id = UUID(user_id)
+
+    # Delete the inspection and get the returned data
+    deleted_inspection = inspection.delete_inspection(cursor, inspection_id, user_id)
```

Comment:
I usually manually check if the user_id is associated with the inspection before calling the function to give better insight to our other layers on the errors

Reply:
Is it all the same if it is checked inside the function? Checking inside the function avoid an additional round trip to the DB.

Reply:
If we check it before hand, we can return a custom error for the BE to have better insight. If you check it within the DB function, I feel like its harder to give context on the error, unless you know an easy way.

Reply:
The easiest way I found was to raise in `delete_inspection` a `InspectionDeleteError` from the db exception that would be raised when the user_id isn't right, passing along all the details.

Check `"fertiscan_0.0.13".delete_inspection`:
```sql
    IF NOT EXISTS (
        SELECT 1
        FROM "fertiscan_0.0.13".inspection 
        WHERE id = p_inspection_id 
        AND inspector_id = p_inspector_id
    ) THEN
        RAISE EXCEPTION 'Inspector % is not the creator of inspection %', p_inspector_id, p_inspection_id;
    END IF;
```

and `inspection.delete_inspection`
```python
    except (Error, DatabaseError, OperationalError) as e:
        raise InspectionDeleteError(f"Database error occurred: {str(e)}") from e
```

That should ensure the exception details bubble all the way up to the BE.